File: Solution/Programmers/이중우선순위큐.py
import heapq

def solution(operations):
    maxheap = []
    minheap = []
    for op in operations:
        cmd, data = op.split()
        if cmd == 'I':
            num = int(data)
            heapq.heappush(minheap, num)
            heapq.heappush(maxheap, (-num, num))
        else:
            if not minheap:
                continue
            if data == '1':
                maxVal = heapq.heappop(maxheap)[1]
                minheap.remove(maxVal)
            else:
                minVal = heapq.heappop(minheap)
                maxheap.remove((-minVal, minVal))
    #print(maxheap, minheap)
    if not maxheap or not minheap:
        return [0, 0]
    else:
        return [maxheap[0][1], minheap[0]]

File: Solution/Programmers/이중우선순위큐.js
function solution(operations) {
  const q = [];
  for (let operation of operations) {
    let [cmd, x] = operation.split(" ");
    x *= 1;
    if (cmd === "I") {
      q.push(x);
    } else {
      if (x === 1) {
        const idx = q.indexOf(Math.max(...q));
        q.splice(idx, 1);
      } else {
        const idx = q.indexOf(Math.min(...q));
        q.splice(idx, 1);
      }
    }
  }

  const ans = [0, 0];
  if (!q.length) return [0, 0];
  ans[0] = Math.max(...q);
  ans[1] = Math.min(...q);

  return ans;
}
